### PR TITLE
Add type hints

### PR DESCRIPTION
**Pull request by greg0ire**
I am targeting this branch, because this is a BC break.

## Changelog

```markdown
### Added
- PHP 7.1 type hints
```

### CODE REVIEW

File: src/Block/FormatterBlockService.php [hunk 1]
```diff
@@ -65,10 +65,14 @@ public function configureSettings(OptionsResolver $resolver): void
         ]);
     }
 
-    public function getBlockMetadata($code = null)
+    public function getBlockMetadata($code = null): Metadata
```
Comment (kunicmarko20) [inline src/Block/FormatterBlockService.php]: Should this maybe be `string` typehint?

File: src/Controller/CkeditorAdminController.php [hunk 2]
```diff
@@ -147,10 +143,7 @@ public function uploadAction()
         ]);
     }
 
-    /**
-     * @return string
-     */
-    private function getTemplate($name)
+    private function getTemplate($name): string
```
Comment (kunicmarko20) [inline src/Controller/CkeditorAdminController.php]: Typehint for name?

File: src/Block/FormatterBlockService.php (hunk 1 above)
Comment (greg0ire) [reply src/Block/FormatterBlockService.php]: No, the parent class does not have it.

File: src/Block/FormatterBlockService.php [hunk 3]
```diff
@@ -65,10 +65,14 @@ public function configureSettings(OptionsResolver $resolver): void
...
     {
-        return new Metadata($this->getName(), (null !== $code ? $code : $this->getName()), false, 'SonataFormatterBundle', [
-            'class' => 'fa fa-file-text-o',
-        ]);
+        return new Metadata(
+            $this->getName(),
+            (null !== $code ? $code : $this->getName()),
```
Comment (kunicmarko20) [inline src/Block/FormatterBlockService.php]: Do you even need paranthesis?

Comment (greg0ire) [reply src/Block/FormatterBlockService.php]: good catch, removed!

**Comment by greg0ire** [comment]
Well?

**Comment by kunicmarko20** [comment]
Thanks @greg0ire !

**Comment by jmontoyaa** [comment]
I have  this error:
```

Compile Error: Declaration of 
Sonata\FormatterBundle\Twig\SecurityPolicyContainerAware::
checkSecurity(array $tags, array $filters, array $functions): void 
must be compatible with 
Twig_Sandbox_SecurityPolicyInterface::checkSecurity($tags, $filters, $functions)
--
```
 
I checked the latest version of twig/twig: 2.4.8:

https://github.com/twigphp/Twig/blob/v2.4.8/lib/Twig/Sandbox/SecurityPolicyInterface.php

And there are no type hints "void", "bool"...

What version of twig I should use?

**Comment by FlorianKromer** [comment]
I have the same error & have made same analysis. Did I miss something to upgrade to 4-x ? 
```
Compile Error: Declaration of Sonata\FormatterBundle\Twig\SecurityPolicyContainerAware::checkSecurity(array $tags, array $filters, array $functions): void must be compatible with Twig_Sandbox_SecurityPolicyInterface::checkSecurity($tags, $filters, $functions)
Compile Error: Declaration of Sonata\FormatterBundle\Twig\SecurityPolicyContainerAware::checkMethodAllowed($obj, string $method): bool must be compatible with Twig_Sandbox_SecurityPolicyInterface::checkMethodAllowed($obj, $method)
```

**Comment by greg0ire** [comment]
No, looks like this is my mistake, and that this part of the code is not covered by unit tests.

**Comment by greg0ire** [comment]
@jmontoyaa @FlorianKromer please review #347

**Pull request merged**